athena/athena/athena/endpoints.py

```
@@ -285,74 +296,69 @@ def feedback_provider(func: Union[
     @app.post("/feedback_suggestions", responses=module_responses)
     @authenticated
     @with_meta
     async def wrapper(
-            exercise: exercise_type,
-            submission: submission_type,
-            isGraded: is_graded_type = Body(True, alias="isGraded"),
-            learner_profile: learner_profile_type = Body(None, alias="learnerProfile"),
-            latest_submission: latest_submission_type = Body(None, alias="latestSubmission"),
-            competencies: competencies_type = Body(None, alias="competencies"),
+            request: FeedbackProviderRequest,
             module_config: module_config_type = Depends(get_dynamic_module_config_factory(module_config_type))):
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)